docs/api-reference/webgl/copy-and-blit.md

```diff
@@ -0,0 +1,163 @@
+
+
+### copyToBuffer(opts: Object) : Buffer (WebGL2)
+
+Reads data from a `Framebuffer` or `Texture` object into A `Buffer` object and returns it. A new `Buffer` object is created when not provided. This method avoids a sync between CPU and GPU as pixel values are copied from GPU texture memory to GPU Buffer memory. This method returns right away without any delays.
+
+A CPU and GPU sync will be triggered when the returned buffer data is read using `buffer.getData()`, but applications can delay this read, which can reduces the delay due to the sync, or the sync can be completely avoided by using the `Buffer` as the source of input to the GPU (either as `ARRAY_BUFFER` or `PIXEL_UNPACK_BUFFER`).
+
+* opts
+
+  Source options
+  * `source` (`Texture` or `Framebuffer`) - This object will be bound and data copied from it.
+  * `x` - (*number*, default: 0) X offset of the area to be copied,
+  * `y` - (*number*, default: 0) Y offset of the area to be copied,
+  * `width` - (*number*, default: framebuffer width) The width of the area to be copied,
+  * `height` - (*number*, default: framebuffer height) The height of the area to be copied,
+  * `format` - (*GLenum*, default: GL.RGBA) The format of the data.
+  * `type` - (*GLenum*, default: type of `buffer` or `UNSIGNED_BYTE`) The type of the data.
+
+  Target options
+  * `buffer` - (*Buffer*) Buffer object, into which data to be copied.
+  * `byteOffset` - (*number*, default: 0) Byte offset from which data should be copied into buffer.
+
+Notes:
+
+* Reading from floating point textures is dependent on an extension both in WebGL1 and WebGL2.
+* When supported, the `{format: GL.RGBA, type: GL.FLOAT, ...}` combination becomes valid for reading from a floating-point color buffer.
+
+This function makes calls to the following WebGL APIs:
+
+[`gl.readPixels`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/readPixels), [`gl.bindFramebuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindFramebuffer), [`gl.bindBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/bindBuffer)
+
+
+### copyToTexture(opts: Object) : Texture
+
+Copies pixels from a `Framebuffer` or `Texture` object into the specified area of a two-dimensional texture image or cube-map texture image. (gl.copyTexImage2D, gl.copyTexSubImage2D and gl.copyTexSubImage3D wrapper)
+
+* opts
+
+  Source options
+  * `source` (`Texture` or `Framebuffer`) - If provided this object will be bound and data copied from it.
+  * `x` (`GLint`, optional, default: 0) - x coordinate of the lower left corner where to start copying.
+  * `y` (`GLint`, optional, default: 0) - y coordinate of the lower left corner where to start copying.
+
+  Target options
+  * `texture` (`Texture`, optional) - If provided this object will be bound and data copied into it, if not provided `target` must be set and data will be copied to currenlty bound `Texture` object.
```

Comment:
API Audit: I am wondering if all these methods should have an overloaded `target` parameter, instead of custom opts parameters.